Firebase.agent.lib.nut

```diff
@@ -592,20 +636,32 @@ class Firebase {
     }
 
     // process the http response accordingly
-    function _processResponse (request,callback) {
+    function _invokeCallback(request, callback) {
         request.sendasync(function(res) {
-            local data = res.body ;
+            local data = res.body;
             try {
-                data = http.jsondecode(data);
+                if (data == "") {
```

Comment:
It would probably be better to rewrite it to:
```
local response = res.body;
local data = null;
if (response && response.len() > 0) {
    data = http.jsondecode(response);
}
```